src/tsconfig.json

```diff
@@ -0,0 +1,4 @@
+{
+  "extends": "../tsconfig",
```

Comment:
Can you help me understand the advantage of having 2 tsconfig files in this particular case?

Reply:
This is tricky :) `tsconfig.json` in the root adds both `src` and `test` directories into the scope so that tests could be typechecked as well. However then the tests are also included in the dist typings. Typescript allows multiple nested configs and the one in `src` is used for building the dist (see `rollup.config.js`) and it includes only the source.

Reply:
That makes sense, thanks.  I'm def not an expert on structuring TS configs!  A couple questions, mostly out of curiosity, but also wondering if there is an established best practice in the TS community.

Is this structure, where the top-level config includes both src and test, and src config includes only src, considered current best practice?

Is it possible to structure the configs "the other way around", that is where the top-level config includes only the `src` dir, and a second test/tsconfig.json _adds_ the `test` dir?  I'm guessing it's not possible due to using ts-node/register in the mocha.opts file, which probably looks for the top-level config by default?

Reply:
Yeah, it's possible to have different configs with different names in the root but then you have to setup your IDE/editor to pick that by default (to typecheck the tests).

I'm not sure that this is considered best practice but TS dev team has been suggesting multiple configs in different directories several times.

If I'm not mistaken `ts-node` uses default typescript compiler API under the hood which respects all available tsconfigs in the file tree. We could try but again we should configure IDE additionally.

Reply:
@TylorS do you have any thoughts on this?

Reply:
I have a pretty similar setup in a few projects and I've seen numerous OSS projects with similar tsconfig setups. I don't know about other editors, but VSCode automatically picks up the closest tsconfig.json to the file that you are actively viewing by traversing the workspace "up" sorta like node module resolution.

Reply:
Cool, thanks for the discussion.  Now that I understand a bit better, the current top-level + extends seems perfectly reasonable.